src/internal/packager/components.go

```diff
@@ -0,0 +1,207 @@
+package packager
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/defenseunicorns/zarf/src/config"
+	"github.com/defenseunicorns/zarf/src/internal/message"
+	"github.com/defenseunicorns/zarf/src/internal/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/pterm/pterm"
+	"gopkg.in/yaml.v2"
+)
+
+const horizontalRule = "───────────────────────────────────────────────────────────────────────────────────────"
+
+func getValidComponents(allComponents []types.ZarfComponent, requestedComponentNames []string) []types.ZarfComponent {
+	var validComponentsList []types.ZarfComponent
+	var orderedKeys []string
+	var choiceComponents []string
+
+	componentGroups := make(map[string][]types.ZarfComponent)
+
+	// Break up components into choice groups
+	for _, component := range allComponents {
+		key := component.Choice
+		// If not a choice group, then use the component name as the key
+		if key == "" {
+			key = component.Name
+		} else {
+			// Otherwise, add the component name to the choice group list for later validation
+			choiceComponents = appendIfNotExists(choiceComponents, component.Name)
+		}
+
+		// Preserve component order
+		orderedKeys = appendIfNotExists(orderedKeys, key)
+
+		// Append the component to the list of components in the group
+		componentGroups[key] = append(componentGroups[key], component)
```

Comment:
Reading through this just made me realize that we don't validate component names are unique within a package. Could you add that to this PR too (or write an issue if you don't think it fits in this PR's scope)?